pkg/providers/v2/cloud.go

```diff
@@ -46,8 +55,50 @@ var _ cloudprovider.Interface = (*cloud)(nil)
 
 // cloud is the AWS v2 implementation of the cloud provider interface
 type cloud struct {
-	creds  *credentials.Credentials
-	region string
+	creds     *credentials.Credentials
+	instances cloudprovider.InstancesV2
+	region    string
+	tagging   awsTagging
```

Comment:
I think there are some implications around tagging that we may not want to introduce yet, let's add this in a follow up PR?

Reply:
yeah sure!